src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -303,13 +309,14 @@ SymbolKind.NetModule or
                 // check whether we can bail out quickly for typing case
                 var inProgressState = state as InProgressState;
 
-                inProgressProject = inProgressState != null ? inProgressState.IntermediateProjects.First().oldState : this.ProjectState;
+                inProgressProject = inProgressState != null ? inProgressState.PendingTranslationSteps.FirstOrDefault().oldState : this.ProjectState;
```

Comment:
This feels like this has a subtle behavior change: before inProgressState meant we either new oldState would exist, or we'd use this.ProjectState. Now if there is no oldState, we won't look at this.ProjectState well probably set this to null?

I suspect the compiler isn't catching the null behavior here with the FirstOrDefault since it's a tuple...

Reply:
(I'm blocking for this reason; I could be wrong but I think this would break things badly if I'm right...)

Reply:
ok.  i'll tweak.

Reply:
does the updated logic now match your expectation